01_fetch/src/fetch_noaa_nerrs.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+import os 
+import numpy as np
+import urllib
+import requests, json
+
+def fetch_metadata(station_id, metadata_outfile):
+    '''fetch metadata for NOAA NERRS station'''
+    metadata_url = f'https://api.tidesandcurrents.noaa.gov/mdapi/prod/webapi/stations/{station_id}/.json'
+    response = requests.get(metadata_url)
+    text = response.text
+    data = json.loads(text)
+    nested_list = pd.json_normalize(data, record_path=['stations'])
+    df = nested_list.stack().reset_index()
+    metadata = df.iloc[1:,1:].rename(columns={df.columns[1]:"NOAA NERRS metadata:",
+                                              df.columns[2]:(station_id)}).replace(".self","", regex=True)
+    metadata.to_csv(metadata_outfile, index=False)
+
+def fetch_noaa_nerrs_data(start_dt, end_dt, datum, station_id, time_zone, product, units, file_format, data_outfile):
+    '''fetch NOAA NERRS data from select station. Change product argument for NOAA data product. (ie. product = current for current data)'''
+    dates = pd.date_range(start=start_dt, end=end_dt, freq='D')
+    dates = dates.strftime('%Y%m%d').astype(int)
+    url = "https://api.tidesandcurrents.noaa.gov/api/prod/datagetter?product=predictions&application=NOS.COOPS.TAC.WL"
+    data_url = f'{url}&begin_date={dates[0]}&end_date={dates[-1]}&datum={datum}&station={station_id}&product={product}&time_zone={time_zone}&units={units}&interval=&format={file_format}'
+    outfile_format = data_outfile.format(start_dt=start_dt, end_dt=end_dt, station_id=station_id, data_outfile=data_outfile, product=product)
+    urllib.request.urlretrieve(data_url, outfile_format)
+
+def main():
+    station_id = '8551762'
+    
+    datum='MLLW'
```

Comment:
What is the difference between the different datum options that they offer. I'm looking [here](https://api.tidesandcurrents.noaa.gov/api/prod/) and seeing MHHW, MHW, MLW, etc. I guess I'm not super familiar with using datum for water level in an oceanic setting, only in terrestrial, I assume it's similar. How much do you think this choice matters?

Reply:
agree - might be nice to add comments to describe the different options and why we chose the one we did (if it isn't obvious)

Reply:
I believe they are standard to specific regions. I think this is something we should confirm with Salme to see which one she is using for the model. Good point, I definitely should add a comment about the different tide datums.